packages/react/src/components/ComboBox/ComboBox.tsx

```diff
@@ -1094,7 +1111,14 @@ const ComboBox = forwardRef(
             {inputValue && (
               <ListBoxSelection
                 clearSelection={() => {
+                  isManualClearingRef.current = true;
+                  setInputValue('');
+                  onChange({ selectedItem: null });
                   selectItem(null);
+                  setTimeout(() => {
+                    isManualClearingRef.current = false;
+                  }, 0);
```

Comment:
Does using a mutable ref together with a `setTimeout` to manage state introduce a race condition given the ordering of events may not be constant?

Also, the fact that the flag reset is scheduled via `setTimeout` means that there is a short window during which events are ignored. If another change happens during that gap, there may be some unexpected behavior here.